### PR TITLE
PORTALS-2922, PORTALS-2927, PORTALS-2871 - Dependency upgrades

### PR DESCRIPTION
**Pull request by nickgros**
I didn't want to upgrade _everything_ but I couldn't otherwise manage to get pnpm to bump only `axios` (which is only a transitive dependency of `nx`), which itself has a transitive dependency on one of the vulnerable packages.

- Upgrade Nx from v17 to v18 (no breaking changes were shipped in v18)
- Upgrade override @types/react to latest version
- Run `pnpm update -r` to bump every package to the latest versions in the specified range
- Omit `variant` prop from SelectProps in DefaultValueField to fix new type error
- Rewrite hook test that used deprecated `ReactDOM.render` to fix new lint error

### CODE REVIEW

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelFormFields/DefaultValueField.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export type DefaultValueFieldProps<T> = {
   onChange: (newValue: T | undefined) => void
   disabled?: boolean
   TextFieldProps?: Omit<TextFieldProps, 'value' | 'onChange' | 'disabled'>
-  SelectProps?: Omit<SelectProps, 'value' | 'onChange' | 'disabled'>
+  SelectProps?: Omit<SelectProps, 'variant' | 'value' | 'onChange' | 'disabled'>
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelFormFields/DefaultValueField.tsx:23]: Fix type error where `variant` was required

File: packages/synapse-react-client/src/components/Authentication/TOTPForm.tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ export default function TOTPForm(props: TOTPFormProps) {
     <Box>
       <MuiOtpInput
         autoFocus
+        placeholder={''}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/Authentication/TOTPForm.tsx:21]: Fix type error, no visual change

File: package.json [hunk 3]
```diff
@@ -15,19 +15,19 @@
...
-    "eslint-plugin-storybook": "^0.6.12",
-    "eslint-plugin-testing-library": "^5.11.0",
+    "eslint-plugin-storybook": "^0.6.15",
+    "eslint-plugin-testing-library": "^5.11.1",
     "husky": "^8.0.3",
-    "lint-staged": "^13.2.3",
-    "nx": "^17.0.2",
-    "rimraf": "^5.0.1",
+    "lint-staged": "^13.3.0",
+    "nx": "^18.0.7",
```
Comment (nickgros) [inline package.json:29]: I bumped nx from v17 to v18 to transitively bump axios and follow-redirects. There were no breaking changes in nx v18

File: package.json [hunk 4]
```diff
@@ -15,19 +15,19 @@
...
-    "eslint-plugin-storybook": "^0.6.12",
-    "eslint-plugin-testing-library": "^5.11.0",
+    "eslint-plugin-storybook": "^0.6.15",
+    "eslint-plugin-testing-library": "^5.11.1",
     "husky": "^8.0.3",
-    "lint-staged": "^13.2.3",
-    "nx": "^17.0.2",
-    "rimraf": "^5.0.1",
+    "lint-staged": "^13.3.0",
+    "nx": "^18.0.8",
```
Comment (nickgros) [inline package.json:29]: I bumped nx from v17 to v18 to transitively bump axios and follow-redirects. There were no breaking changes in nx v18

File: packages/synapse-react-client/src/utils/hooks/useGetInfoFromIds.test.tsx [hunk 5]
```diff
@@ -1,64 +1,34 @@
...
       ids: ['aaa'],
       type: 'USER_PROFILE',
     }
-    act(() => {
-      ReactDOM.render(
-        <SynapseTestContext>
-          <HookWrapper {...props} />
-        </SynapseTestContext>,
-        container,
-      )
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/hooks/useGetInfoFromIds.test.tsx:56]: Got a new lint error about using `ReactDOM.render`, which was deprecated in React 18. Rewrite the test to use `renderHook` from `@testing-library/react`

File: packages/synapse-react-client/public/mockServiceWorker.js [hunk 6]
```diff
@@ -2,7 +2,7 @@
 /* tslint:disable */
 
 /**
- * Mock Service Worker (1.2.2).
+ * Mock Service Worker (1.3.2).
  * @see https://github.com/mswjs/msw
  * - Please do NOT modify this file.
```
Comment (hallieswan) [inline packages/synapse-react-client/public/mockServiceWorker.js:7]: Was this bumped by the msw upgrade? Just curious given the file's request to not modify the file directly!

**Pull request merged**